apps/native/app/src/ui/lib/card/air-discount-card.tsx

```diff
@@ -0,0 +1,154 @@
+  align-items: center;
+  justify-content: space-between;
+  padding: ${({ theme }) => theme.spacing[2]}px;
+  padding-right: ${({ theme }) => theme.spacing[1]}px;
+  padding-bottom: ${({ theme }) => theme.spacing[1]}px;
+`
+
+const Message = styled.View`
+  align-items: flex-start;
+  flex: 1;
+`
+
+const Right = styled.View`
+  align-items: flex-end;
+`
+
+const CopyCodeContainer = styled.View`
+  border-top-width: ${({ theme }) => theme.border.width.standard}px;
+  border-top-color: ${dynamicColor(
+    ({ theme }) => ({
+      light: theme.color.blue200,
+      dark: theme.shades.dark.shade300,
+    }),
+    true,
+  )};
+`
+
+const CopyCodeButton = styled.TouchableOpacity`
+  align-items: center;
+  justify-content: center;
+  padding: ${({ theme }) => theme.spacing[2]}px;
+`
+
+const ValidUntil = styled.Text`
+  border-radius: ${({ theme }) => theme.border.radius.large};
+  background-color: ${dynamicColor(
+    ({ theme }) => ({
+      dark: theme.shades.dark.shade300,
+      light: theme.color.blue100,
+    }),
+    true,
+  )};
+  padding: ${({ theme }) => theme.spacing[1]}px;
+  overflow: hidden;
+  margin-bottom: ${({ theme }) => theme.spacing.smallGutter}px;
+
+  ${font({
+    fontWeight: '600',
+    color: ({ theme }) => ({
+      light: theme.color.blue400,
+      dark: theme.color.blue300,
+    }),
+    fontSize: 12,
+    lineHeight: 16,
+  })};
+`
+
+interface AirDiscountProps {
+  name: string
+  credit?: number | null
+  code?: string | null
+  text?: string | null
+  validUntil?: string | Date | null
+}
+
+export function AirDiscountCard({
+  name,
+  code,
+  text,
+  validUntil,
+  credit,
+}: AirDiscountProps) {
+  const intl = useIntl()
+  const theme = useTheme()
+  const discountCode = credit === 0 ? undefined : code ?? '0'
+  return (
+    <Host>
+      <Content>
+        <Message>
+          <Typography
+            variant="heading4"
+            style={{ marginBottom: theme.spacing[1] }}
+          >
+            {name}
+          </Typography>
+          <Typography
+            style={{
+              paddingRight: theme.spacing[2],
+              marginBottom: theme.spacing[2],
+            }}
+          >
+            {text}
+          </Typography>
+        </Message>
+        <Right>
+          {validUntil && (
+            <ValidUntil>
+              <FormattedMessage
+                id="airDiscount.validTo"
+                values={{
+                  date: intl.formatDate(validUntil),
+                  time: intl.formatTime(validUntil),
+                }}
+              />
+            </ValidUntil>
+          )}
+          <Typography
+            style={{
+              paddingRight: theme.spacing[2],
+              marginBottom: theme.spacing[2],
+            }}
+          >
+            {discountCode}
+          </Typography>
+        </Right>
+      </Content>
+      {credit !== 0 && (
+        <CopyCodeContainer>
+          <CopyCodeButton onPress={() => Clipboard.setString(code ?? '')}>
```

Comment:
Should button be visible if code does not exist?

Reply:
I was following the implementation from `AirDiscountOverview` and the only condition for it to be hidden is if credit is 0, see [here](https://github.com/island-is/island.is/blob/main/libs/service-portal/air-discount/src/screens/AirDiscountOverview/AirDiscountOverview.tsx#L202). 
But I thought about it though, can add if we want 😄